infra/modules/providers/azure/service-plan/variables.tf

```diff
@@ -0,0 +1,47 @@
+variable "name" {
+  description = "(Required) Specifies the human consumable label for this resource."
+  default     = ""
+}
+
+variable "resource_group_name" {
+  description = "(Optional) The name of the resource group in which to create the storage account. Changing this forces a new resource to be created. If omitted, will create a new RG based on the `name` above"
+  default     = ""
+}
+
+variable "resource_group_location" {
+  description = "(Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created. The full list of Azure regions can be found at https://azure.microsoft.com/regions"
+}
+
+variable "resource_tags" {
+  description = "Map of tags to apply to taggable resources in this module.  By default the taggable resources are tagged with the name defined above and this map is merged in"
+  type        = "map"
+  default     = {}
+}
+
+variable "svcplan_name" {
```

Comment:
As mentioned in previous comments, please avoid abbreviating resource names suggestion `service_plan_name`

Reply:
@erikschlegel: Seems like bedrock is not consistent with their approach of naming which is causing all this confusion and additional churn in project cobalt. I am looking at [this](https://github.com/Microsoft/bedrock/blob/42047255990d34b95a359943304f7b5cea066fa7/cluster/azure/backend-state/resource-group.tf) example which was provided by you [before](https://github.com/Microsoft/cobalt/pull/72#discussion_r276665806), as a reference to the changes needed, where they are using the exact same logic as us in cobalt for the naming of resources. If this feels opinionated for cobalt clients, we can go back to the approach that I had previously built without `local.name`.

Reply:
this comment was more related to avoid abbreviating resource names. Generally speaking, naming convention will vary from customer to customer. Let's start simple with supporting a single variable for service plan name and consider other edge cases based on feedback from our customers.